src/HttpHandler/HttpHandlerFactory.php

```diff
@@ -52,11 +55,15 @@ public static function build(ClientInterface $client = null)
             $version = (int) substr(ClientInterface::VERSION, 0, 1);
         }
 
+        if (!$logger) {
+            $logger = ApplicationDefaultCredentials::getDefaultLogger();
+        }
```

Comment:
nit (suggestion) - make this one line
```suggestion
        $logger = $logger ?? ApplicationDefaultCredentials::getDefaultLogger();
```